core/src/com/unciv/ui/screens/newgamescreen/ModCheckboxTable.kt

```diff
@@ -84,7 +85,9 @@ class ModCheckboxTable(
         savedModcheckResult = null
         clear()
         mods.clear()  // We'll regenerate this from checked widgets
-        baseRuleset = RulesetCache[newBaseRuleset] ?: return
+        baseRuleset = RulesetCache[newBaseRuleset]
+            // We're calling this from init, baseRuleset is lateinit, and the mod may have been deleted: Must make sure baseRuleset is initialized
+            ?: return setBaseRuleset(BaseRuleset.Civ_V_GnK.fullName)
```

Comment:
Not good, we've already changed baseRulesetName! This check should be the first thing in the function